components/files/file_list/file_list.py

```diff
@@ -9,13 +9,35 @@
 
 sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "..", "tokens"))
 sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "..", "utils"))
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "..", "metadata", "rdm_parameters", "rdm_pagination"))
+sys.path.append(os.path.join(os.path.dirname(os.path.realpath(__file__)), "..", "..", "components", "metadata", "metadata_list"))
 
 from get_token import *
 from utils import *
 from args import *
+from rdm_pagination_traits import *
+from metadata_list  import *
 
 session = requests.Session()
 
+def query_nested_files_and_folders(a_server_config, a_site_id, a_parent_folder_uuid, a_headers):
+
+    files_and_folders = []
+    page_traits = MetadataPaginationTraits(a_page_size="500", a_start=[a_parent_folder_uuid, "fs::file"], a_end=[a_parent_folder_uuid, "fs::file", None])

```

Comment:
if you want files and folders you can do`a_start=[a_parent_folder_uuid], a_end=[a_parent_folder_uuid, None]`

Reply:
when you're querying nested folders you can do `fi["value"]["_type"] == "sl::folder"`

Reply:
This works with "fs::folder"